includes/class-webmention-receiver.php

```diff
@@ -824,10 +824,43 @@ public static function comment_template_include( $template ) {
 		return $template;
 	}
 
+	/**
+	 * Webmention headers to be added
+	 *
+	 * @return boolean
+	 */
+	public static function receive_mentions() {
+		if ( 0 !== WEBMENTION_ALWAYS_SHOW_HEADERS ) {
+			return true;
+		}
+		if ( ! is_singular() ) {
+			// Cache this as will be called twice in a page load
+			$post_id = wp_cache_get( base64_encode( get_self_link() ), 'wmurl' );
```

Comment:
But this will only work if you have installed a cache plugin: https://codex.wordpress.org/Class_Reference/WP_Object_Cache#Persistent_Caching

Reply:
Must disagree. That is if you want it to persist for multiple page loads. I want it to persist during the same page load...once for the http headers and once for the html headers

Reply:
Ah I see: https://codex.wordpress.org/Class_Reference/WP_Object_Cache#Examples

Reply:
If someone enables a persistent cache, then they'd derive an extra benefit

Reply:
I learned a lot today! thanks!